jlm/hls/backend/rvsdg2rhls/add-forks.cpp

```diff
@@ -37,8 +37,8 @@ add_forks(jlm::rvsdg::region * region)
         add_forks(structnode->subregion(n));
       }
     }
-    // If a node has no inputs it is a constant
-    bool isConstant = node->ninputs() == 0;
+    bool isConstant = rvsdg::is<rvsdg::bitconstant_op>(node->operation())
+                   || rvsdg::is<rvsdg::ctlconstant_op>(node->operation());
```

Comment:
What is the reason for changing this?

Reply:
@sjalander

Reply:
@davidmetz 
Can you explain why this change has been made?

This is how the code is for the addr-queue branch, but the earlier version of the code, which checks for no inputs, also works for the benchmarks that are included in the hls-test-suite.

Reply:
@sjalander @davidmetz

Reply:
@phate 
I discussed with David, and this change was to handle local memory. But further changes has med this unnecessary as the test suite works without it.
So I've reverted the change.